File: Linkedlist/2181. Merge Nodes in Between Zeros/solution.py
# Definition for singly-linked list.
# class ListNode:
#     def __init__(self, val=0, next=None):
#         self.val = val
#         self.next = next
class Solution:
    def mergeNodes(self, head: Optional[ListNode]) -> Optional[ListNode]:
        node = head.next
        prev = head 
        sum =0 
        while(node):
            if(node.val == 0):
                prev = prev.next
                prev.val = sum
                sum =0
            sum += node.val
            node = node.next
        prev.next=None
        return head.next

        
        

File: Linkedlist/2181. Merge Nodes in Between Zeros/solution.cpp
/**
 * Definition for singly-linked list.
 * struct ListNode {
 *     int val;
 *     ListNode *next;
 *     ListNode() : val(0), next(nullptr) {}
 *     ListNode(int x) : val(x), next(nullptr) {}
 *     ListNode(int x, ListNode *next) : val(x), next(next) {}
 * };
 */
class Solution {
public:
    ListNode* mergeNodes(ListNode* head) 
    {
        ListNode* node = head->next;
        ListNode* prev = head;
        int sum =0;
        while(node)
        {
            sum += node->val; 
            if(node->val == 0)
            {
                prev = prev->next;
                prev->val = sum;
                sum = 0;
            }
            node = node->next;
        } 
        prev->next = NULL;  
        return head->next;
    }
};